### DISCUSSION TITLE
Projects (Beta) - Side Panel / Preview for issues

### DISCUSSION THREAD

**Discussion by marty0678** [community/community#5381] [Projects and Issues] [answered]
In the old/Kanban-style board, clicking an issue opens the side panel where you can preview the issue before going to a new tab to see the full issue. Request to have a similar option in Projects Beta, as the current implementation opens each issue in a new tab. Would be nice to be able to see a preview showing the description (and bonus: comments) before having to open a new tab to avoid tabbing around your browser.

**Comment by adam-pheedloop** [comment]
+1

**Reply by rodsfreitas** [reply]
Yeah, this was the first thing that I missed when trying Projects (Beta). The existing one in the old projects is really well done and handy.

**Comment by arohland** [comment]
This is basically the feature we are missing to move from our old project to the new issues!

**Reply by virtualguy** [reply]
we are blocked on this too, and having the comments section would massive

**Comment by nickhagen** [comment]
how do we push up this feature request to the "top of the list" !?

**Reply by rodsfreitas** [reply]
Invite your friends to vote for this :)

**Comment by douglaslassance** [comment]
A sidebar to edit one or more selected issues would be nice, but personally, I would prioritize allowing to display and edit fields on the cards directly.

**Reply by marty0678** [reply]
That would also be a welcome addition for sure.

**Comment by petewalker** [comment]
This seems to have been added to the public roadmap: https://github.com/github/roadmap/issues/287

**Reply by markbullen** [reply]
As of 9th Feb 2022 it has been removed from that roadmap issue and does not currently appear on anywhere on the public roadmap. 

![image](https://user-images.githubusercontent.com/11536573/161236641-5c9fd65a-616e-403a-8f68-d9e0db09c422.png)

**Reply by marty0678** [reply]
Well, that's depressing.

**Reply by petewalker** [reply]
I wouldn't read too much into this. They renamed the issue to be "milestone 1" too. I suspect they've just chunked the features into smaller milestones so they'll fit into a quarter. Hopefully it'll reappear in a future milestone..!

**Comment by lnryan** [comment]
It appears the project team considers this a "nice-to-have" feature given that it's only mentioned in "Future" beyond Sept '22 😱 

I would like to highlight why being able to view the issue **on the same page** is of higher priority and should be handled separately from a catch-all "Parity" ticket when you're asking teams to work with this public Beta and provide feedback.

- For useful feedback, teams need to use this _as_ their project board, not as a novel side experiment. Who has the capacity for that, really?
- Many teams continue to work in a remote capacity. Stand-ups are remote, and revolve around sharing the project board on video calls. 
- Forcing teams to navigate tabs in remote standups in order to see what's in a ticket is awkward and time-consuming
- Forcing teams to navigate tabs in remote standups forces the presenter to adopt a lower-security screensharing option (share screen, not share tab) that then exposes them to the risk of sharing anything else that occurs on their laptop during the call. This is unnecessary in increases the burden on the presenter to close out other work.

Not prioritizing the sidebar view suggests that the github team is not really considering how this new Project Board would be used in agile teams, which is perplexing.

**Reply by cltsang** [reply]
Being in "Future" in the public roadmap doesn't necessarily mean it's beyond Sep 22.
You can see they've been working on features with "Future" status in [github universe video](https://www.youtube.com/watch?v=ha1KHcPMAEk), like hierarchy navigation.
So all hope is not lost.

**Reply by mariorod** [reply]
Correct ... future is mainly a way for us to signal that we have it high in our priority list but do not have a high confidence date yet. You can see by our [frequent releases](https://github.blog/changelog/label/issues/) that we take things from the future bucket and move them into their quarters as we pick them up and ship them.

**Comment by ccolella-mdc** [comment]
This is pretty essential. Especially to use this for project management when some of the people involved have no reason to look at the details of a pull request, for example. 

Plus there needs to be a place to see and edit all of the field values without having to have them all cluttering up the table.

**Comment by writeblankspace** [comment]
I really need this! I'd love the new projects, but the thing that really made me fall in love for the old projects was this sidebar and ease of access so I can see everything from one tab.

**Reply by LosAhuertos** [reply]
Р

**Comment by MetalMagno** [comment]
We need this asap

**Comment by seanonthenet** [comment]
Miss this feature

**Comment by omerbensaadon** [comment]
This is in progress, lookout for this in the next few weeks 👀

**Reply by petewalker** [reply]
Amazing! Is the public roadmap likely to be updated to reflect your activity soon? There's nothing firm in there post Q1 2022.

**Reply by omerbensaadon** [reply]
Updating it as we speak!

Sorry for the delay

**Reply by ccolella-mdc** [reply]
Heck yeah! So looking forward to this!

**Reply by marty0678** [reply]
Amazing! Very much looking forward to this!

**Reply by marty0678** [reply]
This is fantastic! Thank you very much for implementing this, it's very useful!

I noticed it only activates on board views and table views still open a new tab. I'm assuming both table and board views should operate the same way (ideally they would I think, especially because there is the button in the side panel to open the issue in a new tab or you can CTRL+Click to get the old functionality).

**Reply by marty0678** [reply]
Just got a notification (or at least just noticed) that I had a new feature preview to enable the side panel. Once I did the side panel was activated for table view as well, thank you!

**Comment by benjamin-bergia** [comment]
Just noticed it on my Boards. It is really nice but could it be added, maybe as an option, on the table views too? Currently it is enabled on the Board views only.

**Reply by marty0678** [reply]
Just got a notification (or at least just noticed) that I had a new feature preview to enable the side panel. Once I did the side panel was activated for table view as well.

**Comment by zenginerler** [comment]
They released it as a [feature preview](https://github.blog/changelog/2022-05-05-the-new-github-issues-may-5th-update/) 🎉 

You can enable it by clicking:
- **profile picture** (right top corner) > Feature Preview > Projects Side Panel > Enable

<img width="600" alt="image" src="https://user-images.githubusercontent.com/64453575/167338634-14a85e18-ecbc-4700-9df8-4f8a0cda829b.png">

New version looks like this:
<img width="600" alt="image" src="https://user-images.githubusercontent.com/64453575/167352033-cf9463ef-6dff-4382-b927-a412b45a2266.png">

**Comment by humanapp** [comment]
+1. This feature was very useful in v1

**Reply by iarc13** [reply]
@humanapp how to enable this ? My PR's seem to open in new page only.

**Reply by omerbensaadon** [reply]
this doesn't work for PRs, only Issues, sorry for the delayed response

**Comment by omerbensaadon** [comment, marked as answer]
This feature has been released as a feature preview and enabled globally! 

Check out the Changelog where we announced this: https://github.blog/changelog/2022-05-05-the-new-github-issues-may-5th-update/

**Reply by marty0678** [reply]
Been using it for a few weeks now, and it's gotten better over that time too I think. Thank you, great work!

**Reply by FelixVita** [reply]
Works great! But when I'm adding labels in the side-bar view it only shows the top 12-or-so **issue labels**. Would it be possible to have it show _all_ labels?

**Reply by omerbensaadon** [reply]
I'm pretty confident that if you start searching it shows the rest of them

**Reply by FelixVita** [reply]
> I'm pretty confident that if you start searching it shows the rest of them 

This is true, and I should've mentioned that I was aware of this. Speaking for myself and my colleagues (and I'd be willing to guess _most_ people), the way we use labels when creating a new task is that we glance through the list of all 25-or-so labels and pick the ones that are relevant. We don't have them all memorized. 

In other words, the main problem with relying on the label search you mention is that it turns the job of adding task labels into an exercise of mental recall instead of recognition. And human brains really suck at recall.

It's not a huge deal for me personally, as I'm already getting used to the workaround of right-clicking the task and opening it in a new tab (thus bringing me out of GH Projects and into GH Issues where I can see the full list of labels). But it would be nice to not have to keep explaining to the less experienced devs on my team that the list of labels they see when creating a new task is an incomplete list. They're always visibly confused. It's not intuitive 😁

**Reply by omerbensaadon** [reply]
We definitely want to reduce the number of times folks have to exit Projects. I'll be sure to open an Issue for this internally. Thanks for the feedback!

**Reply by stilliard** [reply]
Really awesome feature! Could we add the same for Pull Requests as well? Currently these still open in a new tab.

**Reply by thuan1412** [reply]
Great feature! Could we have this feature on Issues as well?
